resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -975,4 +985,61 @@ public class ApiClient implements AutoCloseable {
         @Override
         public void close() throws Exception { }
     }
+
+     private ApiClientConnectorResponse retryInvoke(ApiClientConnectorRequest connectorRequest, ApiClientConnectorResponse connectorResponse) {
+         int retriesBeforeBackOff = 0;
+         long waitTimeInMillis;
+         int backOffRetries = 1;
+         long totalWaitTimeInMillis = 0L;
+         int MAX_RETRIES_BEFORE_BACKOFF = 5;
+         long FIVE_MINUTES_IN_MILLIS = 300000L;
+         long MAX_BACKOFF_IN_MILLIS = 27000L;
+         int MAX_BACKOFF_RETRIES = 3;
+
+         try {
+             do {
+                 if (connectorResponse.getHeaders().containsKey("Retry-After") && retriesBeforeBackOff++ < MAX_RETRIES_BEFORE_BACKOFF) {
+                     waitTimeInMillis = Integer.parseInt(connectorResponse.getHeaders().get("Retry-After")) * 1000L;
+
+                 } else if (backOffRetries > MAX_BACKOFF_RETRIES) {
+                     waitTimeInMillis = MAX_BACKOFF_IN_MILLIS;
+
+                 } else {
+                     waitTimeInMillis = getWaitTimeExp(backOffRetries);
+
+                     do {
+                         totalWaitTimeInMillis += waitTimeInMillis;
+                         Thread.sleep(waitTimeInMillis);
+                         connectorResponse = connector.invoke(connectorRequest);
+
+                         if (isFiveMinutesOver(totalWaitTimeInMillis, FIVE_MINUTES_IN_MILLIS)) {
+                             waitTimeInMillis = getWaitTimeExp(++backOffRetries);
+                             totalWaitTimeInMillis = 0L;
+                         }
+                     }
+                     while (isRetryable(connectorResponse.getStatusCode()) && backOffRetries <= MAX_BACKOFF_RETRIES);
+                 }
+
+                 if (isRetryable(connectorResponse.getStatusCode())) {
+                     Thread.sleep(waitTimeInMillis);
+                     connectorResponse = connector.invoke(connectorRequest);
+                 }
+             } while (isRetryable(connectorResponse.getStatusCode()));
+
+         } catch (Exception exception) {
```

Comment:
Where may the exception be throw?

Reply:
As per my understanding, we don't want to throw an exception from here instead return the original response.